modin/experimental/core/execution/native/implementations/hdk_on_native/test/test_dataframe.py

```diff
@@ -447,10 +447,7 @@ def applier(lib):
         eval_general(pd, pandas, applier)
 
     @pytest.mark.parametrize("is_multiindex", [True, False])
-    @pytest.mark.parametrize(
-        "column_names", [None, ["level1", None], ["level1", "level2"]]
-    )
-    def test_reset_index_multicolumns(self, is_multiindex, column_names):
```

Comment:
Because, in case of MultiIndex columns, it falls back to Pandas. The test fails, in this mode, due to the forced lazy execution. MultiIndex renaming is not supported for columns, but it's supported for rows. I think, it makes more sense to test this functionality, rather than default pandas functionality.

Reply:
So previously this test used to pass successfully despite the columns being MultiIndex? I wonder then whether it's a proper solution to mark the whole frame as `has_unsupported_data=True` if it's only a thing for the `.rename()` and maybe some other **<i>subset</i>** of functions? Shouldn't we only mark `.rename()` and those other methods that can't handle MultiIndex columns as falling back to pandas and leave the others untouched since they can process this with no problems?

Reply:
The test passed just because it does nothing with columns. In fact, a frame with MultiIndex columns is totally broken due to column names conversion to HDK-supported format. It's not even possible to print such a frame. If we mark only subset of operations, as you suggested, there could be a situation, when some supported operations applied and when an unsupported operation is called, it will not be able to fall back to pandas, because to_pandas() fails for such frames.

After the fix, this test failed due to forced lazy execution. I could set force_lazy=False in the test, but, I think, it does not make any sense. If we have MultiIndex columns, we are testing pandas, but not modin. That's why I've removed the multiindex columns.